fineract-core/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java

```diff
@@ -270,83 +112,167 @@ public boolean getIsColumnIndexed() {
             }
         }
-        return allowed;
+        return false;
     }
 
-    public boolean isColumnValueNotAllowed(final String match) {
-        return !isColumnValueAllowed(match);
+    public boolean isColumnCodeAllowed(final Integer match) {
+        for (final ResultsetColumnValueData allowedValue : this.columnValues) {
+            if (allowedValue.codeMatches(match)) {
+                return true;
+            }
+        }
+        return false;
     }
 
-    public boolean isColumnCodeNotAllowed(final Integer match) {
-        return !isColumnCodeAllowed(match);
+    public boolean hasPrecision(@NotNull DatabaseType dialect) {
+        return columnType.hasPrecision(dialect);
     }
 
-    public boolean isColumnCodeAllowed(final Integer match) {
-        boolean allowed = false;
-        for (final ResultsetColumnValueData allowedValue : this.columnValues) {
-            if (allowedValue.codeMatches(match)) {
-                allowed = true;
+    // --- Calculation ---
+
+    private String adjustColumnType(String type) {
+        type = type.toUpperCase();
+        switch (type) {
+            case "CLOB":
+            case "ENUM":
+            case "SET":
+                return "VARCHAR";
+            case "NEWDECIMAL":
+                return "DECIMAL";
+            case "LONGLONG":
+                return "BIGINT";
+            case "SHORT":
+                return "SMALLINT";
+            case "TINY":
+                return "TINYINT";
+            case "INT24":
+                return "INT";
+            default:
+                return type;
+        }
+    }
+
+    @NotNull
+    private DisplayType calcDisplayType() {
+        DisplayType displayType = null;
+        if (this.columnCode == null) {
+            displayType = calcColumnDisplayType(columnType);
+        } else {
+            if (columnType.isIntegerType()) {
+                displayType = DisplayType.CODELOOKUP;
+            } else if (columnType.isVarcharType()) {
+                displayType = DisplayType.CODEVALUE;
             }
+
         }
-        return allowed;
+        if (displayType == null) {
+            throw new PlatformDataIntegrityException("error.msg.invalid.lookup.type",
+                    "Invalid Lookup Type:" + this.columnType + " - Column Name: " + this.columnName);
+        }
+        return displayType;
     }
 
-    public String getColumnCode() {
-        return this.columnCode;
+    public static DisplayType calcColumnDisplayType(JdbcJavaType columnType) {
+        if (columnType.isTextType()) {
+            return DisplayType.TEXT;
+        }
+        if (columnType.isStringType()) {
+            return DisplayType.STRING;
+        }
+        if (columnType.isAnyIntegerType()) {
+            return DisplayType.INTEGER;
+        }
+        if (columnType.isAnyFloatType()) {
+            return DisplayType.FLOAT;
+        }
+        if (columnType.isDecimalType()) { // Refer org.drizzle.jdbc.internal.mysql.MySQLType.java
```

Comment:
we are not using drizzle anymore.

Reply:
I do not use drizzle, this is an old comment. There are 3 other places in fineract still with this comment.

Reply:
I was wondering whether this comment can be removed then, also whether the isDecimal check is up to date for mysql connector / mariadb connector and postgres connector.

Reply:
The isDecimal check is correct, because JdbcJavaType is dialect independent/aware.
I would not update this PR just for an old code comment removal, but next time when I touch this code I'll take care.